### DISCUSSION TITLE
Has anyone managed to get the example report to work with 2.10.3?

### DISCUSSION THREAD

**Discussion by willappleford** [netbox-community/netbox#5729] [Q&A] [answered]
Hi there,

New to Netbox reports (and GitHub) and trying to get the example report from this address working with 2.10.3:
https://netbox.readthedocs.io/en/stable/additional-features/reports/

The problem I am hitting when I run it is reported as:
AttributeError: 'ConsolePort' object has no attribute 'connection_status'

I see in 2.10.0 release notes that the following was changed:
dcim.ConsolePort:
Replaced connection_status with connected_endpoint_reachable (boolean)

But when I change 'connection_status' to 'connected_endpoint_reachable' I also get:
AttributeError: 'ConsolePort' object has no attribute 'connected_endpoint_reachable'

When I check out the following file:
netbox/netbox/templates/dcim/consoleport.html

...I see it does not use connected_endpoint_reachable either, but in fact uses:
object.path.is_active

Basically, my question is, which should I be using to get this report working in 2.10.3?

Thanks for any assistance...

**Comment by candlerb** [comment]
Quite possibly it hasn't been updated.  Until it's fixed, there are plenty more sample reports you can try here:
https://github.com/netbox-community/reports/tree/master/reports

**Comment by candlerb** [comment, marked as answer]
`connected_endpoint_reachable` is an attribute in the REST API only.  It's defined as:

```
    @swagger_serializer_method(serializer_or_field=serializers.BooleanField)
    def get_connected_endpoint_reachable(self, obj):
        if obj._path is not None:
            return obj._path.is_active
        return None
```

On the Django models, there is `class PathEndpoint` from which the other components inherit.  It has:

```
    @property
    def path(self):
        return self._path

    @property
    def connected_endpoint(self):
        """
        Caching accessor for the attached CablePath's destination (if any)
        """
        if not hasattr(self, '_connected_endpoint'):
            self._connected_endpoint = self._path.destination if self._path else None
        return self._connected_endpoint
```

Therefore, you should replace `power_port.connection_status` with `power_port.path.is_active` - I've tested this and the report now works.

However I'd say this report is rather dumb: it assumes that everything needs two connected power connections, even for devices which have only one power port, or indeed zero power ports (such as a patch panel).

I suggest this instead:

```
    def test_power_connections(self):

        # Check that every active device has all connected power supplies.
        for device in Device.objects.filter(status=DeviceStatusChoices.STATUS_ACTIVE):
            power_ports = 0
            connected_ports = 0
            for power_port in PowerPort.objects.filter(device=device):
                power_ports += 1
                if power_port.connected_endpoint is not None:
                    connected_ports += 1
                    if not power_port.path.is_active:
                        self.log_warning(
                            device,
                            "Power connection for {} marked as planned".format(power_port.name)
                        )
            if connected_ports < power_ports:
                self.log_failure(
                    device,
                    "{} connected power supplies found ({} needed)".format(connected_ports, power_ports)
                )
            else:
                self.log_success(device)
```

**Reply by willappleford** [reply]
Thanks!  That looks sensible.

**Comment by candlerb** [comment]
Raised pull request #5732
